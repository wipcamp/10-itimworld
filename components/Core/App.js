import React from 'react'
import {Layout} from './Global'

export default (Component) => class App extends React.Component {
  render () {
    return (
      <Layout>
        <Component {...this.props} />
      </Layout>
    )
  }
}