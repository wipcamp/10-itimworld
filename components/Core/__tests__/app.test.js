/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../App'
import {Layout} from '../Global'

const Dump = () => <div>Dump</div>

const HOC = App(Dump)

describe('<App />', () => {
  it('App should be renders "<Dump />"', () => {
    const app = shallow(<HOC />)
    expect(app.find(Layout).dive().find('div').text()).toEqual('<Dump />')
  })

  it('App should be renders with Dump', () => {
    const component = renderer.create(<HOC />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
