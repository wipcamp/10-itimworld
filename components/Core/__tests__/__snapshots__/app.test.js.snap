// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App should be renders with Dump 1`] = `
<div
  className="sc-bdVaJa eAwGjB"
>
  <div>
    Dump
  </div>
</div>
`;
